### ISSUE TITLE
Cannot set breakpoints in Meteor builds of node server app

### ISSUE THREAD

**Issue by brucejo75** [Microsoft/vscode#22818]
I connected with @weinand, on my [stackoverflow](http://stackoverflow.com/questions/42640867/cannot-set-breakpoints-in-meteor-node-app-with-vs-code) posting.  Adding issue as requested.

- VSCode Version: 1.10.2
- OS Version: Windows 10 Version 1607 (OS Build 14393.953)
- Meteor: 1.4.3.2 (includes node v4.8)

### Steps to Reproduce:
1. **in VS Code**: configure `launch.json` attach:
```
  {
    "name": "Attach to Server",
    "type": "node",
    "request": "attach",
    "port": 5838,
    "address": "localhost",
    "restart": true,
    "timeout": 20000,
    "sourceMaps": true,
    "protocol": "legacy",
    "outFiles": ["${workspaceRoot}/meris/.meteor/local/build/programs/server/**/*.js"]
  }
```
2. **in CMD Shell:** set `NODE_OPTIONS` environment variable `set NODE_OPTIONS=--debug=5838`
3. **in CMD Shell:** start meteor: `meteor`
4. **in VS Code:** set a breakpoint on a source file clicking in far left column and get bright red indicator.
5. **in VS Code:** attach to the process.  Successfully attaches
6. **in webapp (connected to server):** execute function that will execute code where breakpoint is set.

### Result:
Does not stop debugger at breakpoint set in my source file.

### Other observations:
- if I put a `debugger;` statement into the code
1. The debugger will break on the debugger statement and I can interact with the code using the VS Code debugger;
2. The debugger does not break in my source code, it brings up another tab with the source code in it.  The source code has a sub-title "_read-only inlined content from source map_".  And the file location is at `\1001\`.

- if instead of attaching a VS Code debugger, I attach `node-inspector` everything works as I expect.  After attaching `node-inspector` all of my set breakpoints hit.  No `debugger;` statements required.
1. After step 3 above: **in CMD shell**: `node-debug --debug-port=5838`
2. Breakpoints set in `node-inspector` hit as expected.

- I understand source maps in concept, but I do not understand the format of the file.  But in the sources array, the references to my source files look like this

> `"meteor://💻app/server/main.js"`

It looks a little odd to me... what is with this 💻 character?

- #7117 touches on this problem.  But more research was not performed.  Also, note that meteor now uses node v4.8 instead of  node v0.10.43

### Comments
- Let me know if there are some questions you need answered from the Meteor Development Team.  I will see if I can get them to engage.
- Let me know if you need any other repro information.

**Comment by brucejo75** [comment]
Hi @weinand, @roblourens, any update on this?  Thanks!

**Comment by roblourens** [comment]
Meteor writes sourcemaps with fake `meteor://` paths that the legacy node debugger has trouble resolving to files on disk. Your best shot is probably 

- Changing the meteor config to output actual paths in sourcemaps instead of something like `"meteor://💻app/server/main.js"` (if this is possible)
- Or using a recent Node version and using `"protocol": "inspector"` which is able to deal with the special meteor paths
- Or continue using the workaround you've found already with debugger statements

**Comment by brucejo75** [comment]
Thanks @roblourens, any thoughts on why `node-inspector` is able to handle the fake `meteor://` paths?
I am writing an issue for the meteor folks and want to make sure I understand this.

Thanks!

**Comment by roblourens** [comment]
Because it doesn't need to map the paths to files on disk, which is the hard part. It just takes them as is.

**Comment by brucejo75** [comment]
Thanks @roblourens!

**Comment by teknologist** [comment]
Hi @roblourens Strange thing here....


If I open node-inspector (I eman http://localhost:8080/?port=5858 in Chrome) once, then  I can put break points on vscode even and have it stop there and show debug infos, that is even after closing the node-inspector window completely.

I just need to open http://localhost:8080/debug once, let it load everything and the close it.


What do you think ?

Many thanks in advance

**Comment by roblourens** [comment]
Sure, that could work. I'm not sure what you're asking.

**Comment by teknologist** [comment]
Hi @roblourens 

What I am saying is that I just launch vscode and attach , no breakpoints work.

now if I just launch node-inspector once, let it load then close the window, I go back to VSCode and everything works. I can set breakpoints IN VSCode and I don't need to leave node-inspector open.

So basically I just need to launch node-inspector once at the start of the debugging session and close it immediately (I don't need to keep it open, nor I need to set breakpoints or anything in it))... Then I can work debugging in VSCode for as long as I want.

Previously I didn't need to launch node-inspector at all, it would work out of the box from VSCode.

Hope I am making myself clearer this time....

**Comment by roblourens** [comment]
I see. Can you open a new issue for that? Sounds like it's not related to this one.

**Comment by matthova** [comment]
@teknologist did you make a new issue for this? I'm interested as well.

**Comment by teknologist** [comment]
@matthova Haven't had the time. but I can confirm, I can only set breakpoints in Node inspector.
As soon as I set them in NodeInspector , VSCode stops at the breakpoints and the debug IDE works fine. Breakpoints set in VSCode are plainly ignored.
